src/Hamiltonians/abstract.jl

```diff
@@ -155,6 +155,10 @@ configurations.
 Providing the number `nnzs` of expected calculated matrix elements may improve performance.
 The default estimates for `nnzs` is `dimension(ham)`.
 
+Setting a custom `col_hint` may improve performance for cases where `num_offdiagonals` is 
+not a good estimate of the actual number of nonzero offdiagonal elements in each column. 
+Defaults to `num_offdiagonals(ham, address)`.
```

Comment:
```suggestion
Setting a custom value `col_hint` for the estimated number of nonzero 
off-diagonal matrix elements in each matrix column may improve performance.
The default value for `col_hint` is `num_offdiagonals(ham, address)`.
```
I believe this formulation is a bit clearer.

Note that `build_sparse_matrix_from_LO` is not part of the exported API, so the new keyword argument should also be documented in the docs for `BasisSetRep`.